### PR TITLE
add config + use config for local tests

### PR DESCRIPTION
**Pull request by jonathanadler**

### CODE REVIEW

**Comment by jonathanadler** [comment]
@shlomitc something like this?

File: config/applitools.config.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const { execSync } = require('child_process');
+const merge = require('lodash/merge');
```
Comment (shlomitc) [inline config/applitools.config.js]: You need to move `lodash` to your `dependencies`, it's currently in `devDependencies`

File: config/applitools.config.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+function getPRHeadHash() {
+  const parentsHashArr = getParentsHashArray();
+  const isPullRequest = parentsHashArr.length === 3;
+  const parentHashIndex = isPullRequest
+    ? PULL_REQUEST_PARENT_HASH_INDEX
+    : HEAD_HASH_INDEX;
+  return parentsHashArr[parentHashIndex].trim();
+}
+
+function getBatchId() {
```
Comment (shlomitc) [inline config/applitools.config.js]: On the one hand, we do want a common utility that will eliminate the code duplication with this configuration, but I wonder if it's the `storybook-snapper` responsibility.

Also, `process.env.BUILD_VCS_NUMBER` is an internal CI thing and not something a real opensource will use.

WDYT?

File: src/external-types.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+declare const DATA_READY_HOOK: string;
```
Comment (shlomitc) [inline src/external-types.d.ts]: why is this file required?

File: config/applitools.config.js (hunk 2 above)
Comment (jonathanadler) [reply config/applitools.config.js]: Yeah, I agree.
So what we're saying, is that this config should just extend the applitools' configuration with the `eyes-storybook` configuration.
So the whole file would just be:
```js
module.exports = ({config}) => merge({
  waitBeforeScreenshots: `[${DATA_READY_HOOK}="true"]`,
}, config);
```

Comment (jonathanadler) [reply config/applitools.config.js]: But then it's not that different than my initial solution, no?

File: src/external-types.d.ts (hunk 3 above)
Comment (jonathanadler) [reply src/external-types.d.ts]: Good question 🤔 
I left this by accident

**Comment by jonathanadler** [comment]
@shlomitc 
Updated to what we talked about.
What do you say? should we merge?

File: README.md [hunk 4]
```diff
@@ -50,3 +50,26 @@ visualize('MyComponent', () => {
...
+
+## Testing asynchronious actions
+
+In order for the asynchronious tests to work, `eyes-storybook` must be configured in a way it knows to wait for the component to notify it is done. This is done by setting a flag on the relevant story and inform applitools to look for this flag and wait.
+
+`storybook-snapper` provides a preconfigured `applitools.config.js` which you should use in your own `applitools.config.js` file.
+
+Change your `applitools.config.js` file to the following:
+```js
+const appName = require('./package').name;
```
Comment (shlomitc) [inline README.md]: is this actually related to async tests?
shouldn't this be part of the default configs?

File: README.md [hunk 5]
```diff
@@ -50,3 +50,26 @@ visualize('MyComponent', () => {
...
+const applitoolsConfig = require('storybook-snapper/config/applitools.config');
+
+// optional local configuration file for overrides
+let config;
+
+try {
+  // for local testing you can add the `apiKey` to your private configuration file  
+  config = require('./applitools.private.config.js');
```
Comment (shlomitc) [inline README.md]: not related to async testing

File: package.json [hunk 6]
```diff
@@ -11,7 +11,8 @@
   "types": "dist/src/index.d.ts",
   "files": [
     "dist",
-    "src"
+    "src",
+    "config"
```
Comment (shlomitc) [inline package.json]: Not sure I have a strong opinion about this one.
It might be confusing that this will not be transpiled while `dist` is. Also, now the folder becomes a public API, meaning you will not be able to change it in the future.` index.js`-like is very helpful in these cases or even solutions like `import-path`.

What other libraries are doing in their build steps?

**Comment by jonathanadler** [comment]
@shlomitc 
Regarding your last three comments.
I'm not sure I understand.
If we decide to leave the config file as is, then we need to have a documentation for how to use it

File: README.md (hunk 4 above)
Comment (jonathanadler) [reply README.md]: I didn't want to start searching for the user's `package.json` file from within this library (like Yoshi does, for instance), so I thought it was better to get it from the users.
But we can use [`find-pkg`](https://www.npmjs.com/package/find-pkg), which is what Yoshi does.

File: README.md (hunk 5 above)
Comment (jonathanadler) [reply README.md]: No, but if we leave the config file as we talked about earlier, then we might want to document it, no?

File: package.json (hunk 6 above)
Comment (jonathanadler) [reply package.json]: Again, if we want users to use our preconfigured config file, then we should publish it, shouldn't we?
I went with Yoshi's convention of having config files in a config folder.

File: README.md (hunk 5 above)
Comment (shlomitc) [reply README.md]: so document it correctly. currently, it is sitting under the "async testing" topic which is not related.

File: README.md (hunk 4 above)
Comment (shlomitc) [reply README.md]: is this mandatory for the tool to work? if so, it should be built-in. if not - then why do we bother doing it?

File: README.md (hunk 5 above)
Comment (jonathanadler) [reply README.md]: Oh, I see.
Cool 👍

File: package.json (hunk 6 above)
Comment (shlomitc) [reply package.json]: `yoshi` doesn't have "build" steps, but we currently do because we are transpiling JSX code. so in the `yoshi` case, it is classic that the folder structure is the tool APIs.

Anyway, that's just something I had in mind for a while and don't have a best practice for

File: README.md (hunk 5 above)
Comment (jonathanadler) [reply README.md]: Wait, actually, it **is** related to async testing.
The only reason we need to provide a configuration file is to have async tests.
We offer a preconfigured configuration file in order for the async tests to work.

File: package.json (hunk 6 above)
Comment (jonathanadler) [reply package.json]: I'm not sure I understand, are you talking about `src` or `config`?

Comment (shlomitc) [reply package.json]: config

File: README.md (hunk 5 above)
Comment (shlomitc) [reply README.md]: so it sounds to me like a general prerequisite. something that should come right after the installation part

File: package.json (hunk 6 above)
Comment (jonathanadler) [reply package.json]: I think for the consumer it won't really matter.
Also, I think the overhead of writing a file just for that, and publishing it (i.e. `import-path`-like) isn't worth it at this time.

**Comment by jonathanadler** [comment]
@shlomitc so what do you say?
Shall we merge?
